diff --git a/drivers/clk/imx/clk.c b/drivers/clk/imx/clk.c
index 6c0029950d38..7f877a775ade 100644
--- a/drivers/clk/imx/clk.c
+++ b/drivers/clk/imx/clk.c
@@ -148,6 +148,14 @@ __setup_param("earlycon", imx_keep_uart_earlycon,
 __setup_param("earlyprintk", imx_keep_uart_earlyprintk,
              imx_keep_uart_clocks_param, 0);
 
+static int _arcelik_uart_hack;
+static int __init arcelik_uart_hack(char *par)
+{
+       _arcelik_uart_hack = 1;
+       return 0;
+}
+early_param("arcelik_uart_hack", arcelik_uart_hack);
+
 static void imx_earlycon_uart_clks_onoff(bool is_on)
 {
        struct clk *uart_clk;
@@ -167,7 +175,7 @@ static void imx_earlycon_uart_clks_onoff(bool is_on)
 
                if (is_on)
                        clk_prepare_enable(uart_clk);
-               else
+               else if (!_arcelik_uart_hack)
                        clk_disable_unprepare(uart_clk);
        } while (true);
